source/magpie/group.d

```diff
@@ -528,31 +607,50 @@ public:
         {
             assert(elements.data.length == elementCountTill[pos[0] + 1] - elementCountTill[pos[0]],
                 "Size of data doesn't match size of group column");
-            static foreach(i; 0.. GrpRowType.length)
+
+            void mixinAux(ptrdiff_t si = -1)(size_t ri = 0) @property
             {
-                if(i == pos[1])
+                static if(si > -1)
                 {
-                    foreach(j; elementCountTill[pos[0]] .. elementCountTill[pos[0] + 1])
-                    {
-                        mixin("data[i][j] " ~ op ~ "= elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[i]);");
-                    }
+                    alias i = si;
+                    enum size_t typepos = si;
+                }
+                else
+                {
+                    size_t i = ri;
+                    enum size_t typepos = 0;
+                }
+
+                foreach(j; elementCountTill[pos[0]] .. elementCountTill[pos[0] + 1])
+                {
+                    mixin("data[i][j] " ~ op ~ "= elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[typepos]);");
                 }
             }
+            
+            mixin auxDispatch!(mixinAux);
+            auxDispatch(pos[1]);
         }
         else static if(T.length == 1 && isArray!(T[0]))
```

Comment:
The only thing different between this branch and the one above is 
```diff
-mixin("data[i][j] " ~ op ~ "= elements.data[j - elementCountTill[pos[0]]];");
+mixin("data[i][j] " ~ op ~ "= elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[typepos]);");
```
you should merge them.